src/main/java/net/snowflake/ingest/streaming/internal/IcebergParquetValueParser.java

```diff
@@ -115,7 +157,14 @@ static ParquetBufferValue parseColumnValueToParquet(
         throw new SFException(
```

Comment:
is the isRepetition check needed/applicable for primitive types too? If not it should be in an if !isPrimitive block?

We can structure this method as:
```
if (!statsMap.containsKey()) {
    throw;
}

if (primitive) {
    if (!null) {

    } else {

    }
} else {
    if (!null) {
        return getGroupValue();
    } else {
        if (isRepetition) {
            throw
        }
    }
}
```

Reply:
The `isRepetition` should be checked for both primitive type and group type. As there might be required primitive column with null value which is not allowed.